octomap_server/src/OctomapServer.cpp

```diff
@@ -493,13 +539,13 @@ void OctomapServer::insertScan(const tf::Point& sensorOriginTf, const PCLPointCl
   // mark free cells only if not seen occupied in this cloud
   for(KeySet::iterator it = free_cells.begin(), end=free_cells.end(); it!= end; ++it){
     if (occupied_cells.find(*it) == occupied_cells.end()){
-      m_octree->updateNode(*it, false);
+      m_octree->updateNode(*it, logodds(free_cell_probs.at(*it)), false);
     }
   }
 
   // now mark all occupied cells:
   for (KeySet::iterator it = occupied_cells.begin(), end=occupied_cells.end(); it!= end; it++) {
-    m_octree->updateNode(*it, true);
+    m_octree->updateNode(*it, logodds(occ_cell_probs.at(*it)), false);
```

Comment:
I think this should be _m_octree->updateNode(*it, logodds(occ_cell_probs.at(*it)), **true**);_

Reply:
false is correct (technically it has a default value, so it could be removed like the previous one):
virtual NODE* updateNode(const OcTreeKey& key, float log_odds_update, bool lazy_eval = false);

The previous one was updateNode:
virtual NODE* updateNode(const OcTreeKey& key, bool occupied, bool lazy_eval = false);

Reply:
Method parameter overloading is fun!